rest-api/openapi/custom.yaml

```diff
@@ -0,0 +1,317 @@
+          required: true
+          description: Unique name of the project
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A list of open diagram editors
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/DiagramEditor'
+        '404':
+          description: Project not found
+  /projects/{project_name}/diagrams:
+    get:
+      tags:
+        - Diagrams
+      summary: Get a list of all diagrams for a project by name
+      operationId: getDiagramsByProjectName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A list of diagrams
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Diagram'
+        '404':
+          description: Project not found
+components:
+  schemas:
+    Diagram:
+      type: object
+      description: A diagram
+      properties:
+        id:
+          type: string
+          description: Unique identifier of the diagram
+        name:
+          type: string
+          description: Name of the diagram
+        description:
+          type: string
+          description: Description of the diagram
+        documentation:
+          type: string
+          description: Documentation of the diagram
+    DiagramEditor:
+      type: object
+      description: A diagram editor in Capella
+      properties:
+        id:
+          type: string
+          description: Unique identifier of the diagram
+        name:
+          type: string
+          description: Name of the diagram
+        uri:
+          type: string
+          description: URI of the diagram
+    EObject:
+      type: object
+      description: An abstract model object
+      properties:
+        DUMMY:
+          type: string
+          description: Dummy property
+    Element:
+      type: object
+      description: An element
+      allOf:
+        - $ref: '#/components/schemas/EObject'
+    ExtensibleElement:
+      type: object
+      description: An extensible element
+      allOf:
+        - $ref: '#/components/schemas/Element'
+    ModelElement:
+      type: object
+      description: "Common base for all capella elements\r\n[source: capella study]\n\
+        \nUsed in levels: operational,system,logical,physical,epbs"
+      allOf:
+      - $ref: '#/components/schemas/ExtensibleElement'
+      properties:
+        id:
+          type: string
+          description: "The unique identifier for this element\r\n[source: capella\
+            \ study]"
+        sid:
+          type: string
+          description: The unique system identifier for this element
```

Comment:
If I'm reading this right, all of these type definitions are also now unused.

Reply:
removed these components